lib/pagepile_api.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+class PagePileApi
+  def get_titles(pileid)
+    response = pagepile.get query_url(pileid)
+    title_data = Oj.load(response.body)
+    title_data
+  rescue StandardError => e
+    raise e unless Errno::EHOSTUNREACH.include?(e.class)
+    return {}
+  end
+
+  def page_titles_for_pileid(pileid)
+    titles_response = get_titles(pileid)
+    return [] if titles_response.empty?
+    titles = titles_response['pages']
+    binding.pry
```

Comment:
This is breaking the build.

Reply:
Yes I wanted to ask about this, what if the id entered has no results

Reply:
For the test case, I have tested for a id which gives results, what be the behavior when there is no ouput? Show a error sign saying no results found?

Reply:
I don't know! I suggest testing with an invalid ID, like some very high number, and see what happens.